contracts/src/mixins/ERC4337Mixin.sol

```diff
@@ -0,0 +1,244 @@
+            (bool success, ) = payable(msg.sender).call{
+                value: missingAccountFunds
+            }("");
+            (success);
+        }
+
+        return _packValidationData(false, 0, 0);
+    }
+
+    function _verifySafeTxHash(
+        bytes calldata callData
+    ) private pure returns (bytes32, address, bytes32, bytes32) {
+        (
+            bytes32 safeTxHash,
+            address safeAddress,
+            uint256 chainId,
+            uint256 nonce,
+            address to,
+            uint256 value,
+            bytes memory data,
+            uint8 operation,
+            uint256 safeTxGas,
+            uint256 baseGas,
+            uint256 gasPrice,
+            address gasToken,
+            address refundReceiver,
+            address signer,
+            bytes32 r,
+            bytes32 vs
+        ) = abi.decode(
+                callData,
+                (
+                    bytes32,
+                    address,
+                    uint256,
+                    uint256,
+                    address,
+                    uint256,
+                    bytes,
+                    uint8,
+                    uint256,
+                    uint256,
+                    uint256,
+                    address,
+                    address,
+                    address,
+                    bytes32,
+                    bytes32
+                )
+            );
+        bytes32 computedSafeTxHash = CoreLib._computeSafeTxHash(
+            safeAddress,
+            chainId,
+            nonce,
+            to,
+            value,
+            data,
+            operation,
+            safeTxGas,
+            baseGas,
+            gasPrice,
+            gasToken,
+            refundReceiver
+        );
+
+        require(
+            computedSafeTxHash == safeTxHash,
+            UnexpectedSafeTxHash(computedSafeTxHash)
+        );
+        return (safeTxHash, signer, r, vs);
+    }
+
+    function _verifySignature(
+        bytes32 safeTxHash,
+        bytes calldata signature,
+        address signer,
+        bytes32 r,
+        bytes32 vs
+    ) private pure {
+        (
+            address recoveredSigner,
+            bytes32 extractedR,
+            bytes32 extractedVS
+        ) = CoreLib._recoverSigner(safeTxHash, signature);
+        require(signer == recoveredSigner, UnexpectedSigner(signer));
+        require(r == extractedR, UnexpectedSignatureR(r));
+        require(vs == extractedVS, UnexpectedSignatureVS(vs));
+    }
+
+    function storeTransaction(
+        bytes32 safeTxHash,
+        address safeAddress,
+        uint256 chainId,
+        uint256 nonce,
+        address to,
+        uint256 value,
+        bytes calldata data,
+        uint8 operation,
+        uint256 safeTxGas,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address refundReceiver,
+        address signer,
```

Comment:
TODO: missing comment

`safeTxHash`, `signer`, `r`, `vs` are included here to avoid that these have to be calculated/ verified again. Probably can be optimized.